ChronoKeeper/Storyteller.h

```diff
@@ -0,0 +1,40 @@
+#ifndef STRORYTELLER_CLIENT_H
+#define STROYTELLER_CLIENT_H
+
+
+namespace chronolog
+
+
+
+template <typename ClientId,
+	  typename ChronoTime,
+	  typename StoryAccessToken,
+          typename StoryMetadata,
+          typename RPCChannel>
+class Storyteller
+{
+public:
+	Storyteller( ClientId const&);
+	~Storyteller();
+
+	StoryAccessToken const & getStoryAccessToken ( StoryId const&) const;
+	ChronoTime	const& getStorytellerTime (ClientId const&) const;
+	void 		updateStorytellerTime ( ChronoTime cosnt&);
+	StoryMetadata const&  beginStoryRecording( StoryId const&, StoryAccessToken cosnt&);
+	StoryMetadata const&  endStoryRecording( StoryId const&);
+ 
+private:
+	ClientId clientId; 
+	ChronoTime clientTime;  
+	RPCChannel rpcChannel;
+	std::map < StoryId, std::pair<StoryAccessToken, StoryMetadata> > activeStories;
```

Comment:
You should use ChronicleId+StoryId instead as keys. If it is a shared data structure, there should be a lock to avoid races.

Reply:
> You should use ChronicleId+StoryId instead as keys. If it is a shared data structure, there should be a lock to avoid races.

I think StoryID is generated from ChronicleName+StoryName. So it already has it covered unless you want to have a multi-key map.

Reply:
Storyteller.h is not referenced anywhere int the code.  And going forward I believe we should be using unique StoryIds and ChronicleIds independent of each other. In any case this topic will be revisited as we progress into the ChronoKeeper design.